File: 1200/2174A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while(t-- > 0)
        {
            String a = sc.next();
            String b = sc.next();

            int freq[] = new int[26];
            for(int i=0;i<b.length();i++)
                freq[b.charAt(i)-'a']++;

            boolean flag = true;
            for(int i=0;i<a.length();i++)
            {
                if(freq[a.charAt(i)-'a'] == 0)
                {
                    flag = false;
                    break;
                }

                freq[a.charAt(i)-'a']--;
            }

            if(flag == false)
            {
                System.out.println("Impossible");
                continue;
            }

            String ans = "";
            int ptr = 0;

            for(int i=0;i<26;i++)
            {
                while(ptr < a.length() && (a.charAt(ptr)-'a' <= i))
                {
                    ans += a.charAt(ptr);
                    ptr++;
                }

                while(freq[i] > 0)
                {
                    ans += (char)('a'+i);
                    freq[i]--;
                }
            }

            System.out.println(ans);
        }
    }
}

File: 1200/2174A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        string a, b;
        cin >> a >> b;

        vector<int> freq(26, 0);
        for (char ch : b)
            freq[ch - 'a']++;

        bool flag = true;
        for (char ch : a) 
        {
            if (freq[ch - 'a'] == 0) 
            {
                flag = false;
                break;
            }

            freq[ch - 'a']--;
        }

        if (!flag) 
        {
            cout << "Impossible\n";
            continue;
        }

        string ans = "";
        int ptr = 0;

        for (int i = 0; i < 26; i++) 
        {
            // Append characters from 'a' to 'z' that are less than or equal to the current character
            // in the same order as they appear in 'a'
            while (ptr < (int)a.size() && (a[ptr] - 'a' <= i)) 
            {
                ans += a[ptr];
                ptr++;
            }

            // Append remaining characters for the current character
            while (freq[i] > 0) 
            {
                ans += char('a' + i);
                freq[i]--;
            }
        }

        cout << ans << "\n";
    }

    return 0;
}
